src/sentry/models/avatar.py

```diff
@@ -99,7 +99,10 @@ def save_avatar(cls, relation, type, avatar=None, filename=None):
                 router.db_for_write(File),
             )
         ):
-            instance, created = cls.objects.get_or_create(**relation)
+            if relation.get("sentry_app") and color is not None:
+                instance, created = cls.objects.get_or_create(**relation, color=color)
```

Comment:
Just checking, this will enforce only 2 DB rows per sentry app right? 1 colourful and 1 not?

Reply:
👍  yep it does, lgtm!

Reply:
Yep! Discussed on Slack but putting here for anyone else - color is a boolean so if an avatar with color already exists, it'll overwrite it, and if not, it creates one, and the same goes for no color. So there can only be 2 avatars per sentry app.